inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -187,15 +187,12 @@ std::shared_ptr<ICNNNetwork> V10Parser::parse(const pugi::xml_node& root, std::i
     }
 
     ::ngraph::op::GenericIE::DisableReshape noReshape(allNodes);
-    auto function = std::make_shared<ngraph::Function>(result_nodes, parameter_nodes, GetStrAttr(root, "name", ""));
-    if (!result_nodes.empty()) {
-        for (const auto& assign : assign_nodes) {
-            assign->add_control_dependency(variable_id_to_read_value.at(assign->get_variable_id()));
-            // often Assign node is a leaf of the graph, we add control_dependency for one of the results
-            // to make Assign node visible for traversals get_ops(), get_ordered_ops()
-            result_nodes[0]->add_control_dependency(assign);
-        }
+    auto function = std::make_shared<ngraph::Function>(result_nodes, assign_nodes, parameter_nodes, GetStrAttr(root, "name", ""));
+    for (const auto& assign : assign_nodes) {
+        assign->add_control_dependency(
```

Comment:
What if user creates nGraph function from the code?
Do we have a documentation that after function creation we need to add control dependency?

Reply:
Good question. I belive we have no such documentation, @itikhono right? Do we have description how to create ngraph Function anywhere?

Reply:
Thanks for noticing, I think it was missed in the docs.
I will create a task in GenerilizeSinks epic to cover it.

Reply:
Task 41736 in jira